### PR TITLE
2 5 5 release notes

### PR DESCRIPTION
**Pull request by nathan-px**

### CODE REVIEW

File: content/reference/CLI/updating-volumes.md [hunk 1]
```diff
@@ -427,3 +427,18 @@ Update Volume: Volume update successful for volume exampleVolume
 Warning: Updating group field will not affect the replica placement of already provisioned volumes.
 ```
 
+## extend a sharedV4 volume to non-Portworx Kubernetes nodes
+
+<!-- export or expose? -->
```
Comment (nathan-px) [inline content/reference/CLI/updating-volumes.md:432]: @adityadani some questions here:

File: content/reference/release-notes/portworx.md [hunk 2]
```diff
@@ -6,6 +6,31 @@ keywords: portworx, release notes
...
+
+|**Issue Number**|**Issue Description**|
+|----|----|
+| PWX-14333 | When both the `multipathd` device and the backing device contain the KVDB volume label, Portworx sometimes mistakenly picked up the backing device and failed to mount it. <br/><br/>**User impact:** <!-- not sure --><br/><br/>**Resolution:** Portworx now chooses the `multipathd` device when both `multipathd` and the backing device contain the KVDB volume label. |
+| PWX-14399 | A bug introduced in Portworx version 2.5.0 required manual intervention <!-- what intervention exactly?-->in order to perform a KVDB recovery from dump files. <br/><br/>**User Impact:** <!-- what did users have to do? --><br/><br/>**Resolution:** You can now recover your KVDB from dump files with a single command. |
+| PWX-14164 | Prior to this version, the `pxctl cloudsnap list -a` command sometimes failed to return all cloudsnaps in an objectstore. <br/><br/>**Resolution:** Portworx now returns all cloudsnaps in an objectstore. |
+| PWX-14160 | Fixed an issue in PX SDK where it was was not returning disks and pools in the response. <!-- more description, user impact, and resolution would be helpful here. -->|
```
Comment (nathan-px) [inline content/reference/release-notes/portworx.md:31]: @harsh-px

File: content/reference/CLI/updating-volumes.md (hunk 1 above)
Comment (adityadani) [reply content/reference/CLI/updating-volumes.md:434]: You can rename the topic to `Accessing a sharedv4 volume outside of a Kubernetes cluster`

File: content/reference/release-notes/portworx.md [hunk 3]
```diff
@@ -6,6 +6,32 @@ keywords: portworx, release notes
...
+July 15, 2020
+
+### Improvements
+
+Portworx has upgraded or enhanced functionality in the following areas:
+
+| **Improvement Number** | **Improvement Description** |
+|----|----|
+| PWX-14102 | You can now export a sharedV4 volume outside of a Portworx cluster using specific IP addresses. |
+| PWX-12151 | You can now add labels to a sharedv4 volume so that the remote mount/client uses NFS version 4.2, for example: ```pxctl volume update <vol_name> --label nfs_v4=true``` |
```
Comment (vilasd) [inline content/reference/release-notes/portworx.md:20]: just say nfsv4 and add link to doc

File: content/reference/release-notes/portworx.md [hunk 4]
```diff
@@ -6,6 +6,32 @@ keywords: portworx, release notes
...
+
+### Fixes
+
+The following issues have been fixed:
+
+|**Issue Number**|**Issue Description**|
+|----|----|
+| PWX-14333 | When both the `multipathd` device and the backing device contain the KVDB volume label, Portworx sometimes mistakenly picked up the backing device and failed to mount it. <br/><br/>**User impact:** <!-- not sure --><br/><br/>**Resolution:** Portworx now chooses the `multipathd` device when both `multipathd` and the backing device contain the KVDB volume label. |
+| PWX-14399 | A bug introduced in Portworx version 2.5.0 required manual intervention <!-- what intervention exactly?-->in order to perform a KVDB recovery from dump files. <br/><br/>**User Impact:** <!-- what did users have to do? --><br/><br/>**Resolution:** You can now recover your KVDB from dump files with a single command. |
```
Comment (vilasd) [inline content/reference/release-notes/portworx.md:29]: @adityadani do we need to RL this ?

File: content/reference/release-notes/portworx.md [hunk 5]
```diff
@@ -6,6 +6,32 @@ keywords: portworx, release notes
...
+| PWX-14102 | You can now export a sharedV4 volume outside of a Portworx cluster using specific IP addresses. |
+| PWX-12151 | You can now add labels to a sharedv4 volume so that the remote mount/client uses NFS version 4.2, for example: ```pxctl volume update <vol_name> --label nfs_v4=true``` |
+
+### Fixes
+
+The following issues have been fixed:
+
+|**Issue Number**|**Issue Description**|
+|----|----|
+| PWX-14333 | When both the `multipathd` device and the backing device contain the KVDB volume label, Portworx sometimes mistakenly picked up the backing device and failed to mount it. <br/><br/>**User impact:** <!-- not sure --><br/><br/>**Resolution:** Portworx now chooses the `multipathd` device when both `multipathd` and the backing device contain the KVDB volume label. |
```
Comment (nathan-px) [inline content/reference/release-notes/portworx.md:28]: @adityadani need user impact here

File: content/reference/release-notes/portworx.md (hunk 4 above)
Comment (nathan-px) [inline content/reference/release-notes/portworx.md:29]: @adityadani and here

File: content/reference/CLI/updating-volumes.md [hunk 6]
```diff
@@ -427,3 +429,26 @@ Update Volume: Volume update successful for volume exampleVolume
 Warning: Updating group field will not affect the replica placement of already provisioned volumes.
 ```
 
+## Extend a sharedV4 volume to non-Portworx Kubernetes nodes
+
+<!-- export or expose? -->
+
+By default, sharedV4 volumes exist within a Portworx cluster and cannot be accessed outside that cluster. If you need to access a sharedV4 volume outside of your Portworx cluster, for example, if your Portworx cluster sits on a subset of your Kubernetes nodes, you can extend it.
```
Comment (adityadani) [inline content/reference/CLI/updating-volumes.md:436]:
```
By default, sharedV4 volumes can be accessed only within the Portworx cluster. However you may need to access a sharedV4 volume outside of your Portworx+Kubernetes cluster, for example, a traditional non K8s application running on a VM needs to access data from a K8s app running in the Kubernetes cluster.
 
```

File: content/reference/CLI/updating-volumes.md [hunk 7]
```diff
@@ -427,3 +429,26 @@ Update Volume: Volume update successful for volume exampleVolume
...
 
+## Extend a sharedV4 volume to non-Portworx Kubernetes nodes
+
+<!-- export or expose? -->
+
+By default, sharedV4 volumes exist within a Portworx cluster and cannot be accessed outside that cluster. If you need to access a sharedV4 volume outside of your Portworx cluster, for example, if your Portworx cluster sits on a subset of your Kubernetes nodes, you can extend it. 
+
+<!-- Need a good example of some use cases  -->
+
+To extend a sharedV4 volume to a larger Kubernetes cluster, add the `allow_ips` label to the volume you wish to expose, specifying a comma separated list of IP addresses of non-portworx Kubernetes nodes you wish to mount your sharedV4 volume to:
```
Comment (adityadani) [inline content/reference/CLI/updating-volumes.md:440]: To access a sharedv4 volume outside of the Kubernetes cluster, add the `allow_ips` label to the volume you wish to export....

File: content/reference/CLI/updating-volumes.md [hunk 8]
```diff
@@ -427,3 +429,26 @@ Update Volume: Volume update successful for volume exampleVolume
...
+
+<!-- what's the actual full command here? it seems like the instructions would have this be a kubectl thing, not pxctl. -->
```
Comment (adityadani) [inline content/reference/CLI/updating-volumes.md:442]: Lets divide this into two sections:
1. Create a volume that can be exposed outside the Kubernetes cluster
```
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: px-sharedv4
provisioner: kubernetes.io/portworx-volume
parameters:
  repl: "3"
  sharedv4: "true"
  allow_ips: "<external-node-ip1>,<external-node-ip2>"
allowVolumeExpansion: true
```

2. To export an existing volume use the following pxctl command

```
pxctl volume update <vol_name> --label allow_ips=<external-node-ip1>,<external-node-ip2>
```

File: content/reference/CLI/updating-volumes.md [hunk 9]
```diff
@@ -427,3 +429,26 @@ Update Volume: Volume update successful for volume exampleVolume
...
+
+```text
+pxctl volume update <vol_name> --label allow_ips=<Kubernetes-IP-1>,<Kubernetes-IP-2>
+```
+
+## Enable NFSv4 for a volume
```
Comment (adityadani) [inline content/reference/CLI/updating-volumes.md:448]: Enable NFSv4 for a sharedv4 volume

By default sharedv4 volume uses the NFS v3 protocol. You can instruct Portworx to use NFS v4 for a specific sharedv4 volume by adding the following label.

Create a sharedv4 volume that uses NFS v4 protocol
```
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: px-sharedv4
provisioner: kubernetes.io/portworx-volume
parameters:
  repl: "3"
  sharedv4: "true"
  nfs_v4: "true"
allowVolumeExpansion: true
```

2. To update an existing volume to use NFS v4 protocol use the following pxctl command
pxctl volume update <vol_name> --label nfs_v4=true

File: content/reference/release-notes/portworx.md (hunk 2 above)
Comment (harsh-px) [reply content/reference/release-notes/portworx.md:31]: **User impact**: Users of the [PX SDK](https://docs.portworx.com/reference/developer-sdk/) or `pxctl -j cluster list` would see empty `Disks` and `Pools` fields in the response. 

**Resolution**: A bug was fixed in the PX SDK handler that now populates these 2 missing fields.

File: content/reference/release-notes/portworx.md (hunk 5 above)
Comment (adityadani) [reply content/reference/release-notes/portworx.md:28]: ...failed to mount it. This caused Portworx to not come up on the nodes where internal kvdb was running.

File: content/reference/release-notes/portworx.md (hunk 4 above)
Comment (adityadani) [reply content/reference/release-notes/portworx.md:29]: Yeah lets release not this.

Comment (adityadani) [reply content/reference/release-notes/portworx.md:29]: A bug introduced in Portworx version 2.5.0, broke the documented procedure to perform a KVDB recovery from dump files. A user would have to manually update the config maps and change the labels on the disk to perform the recovery**Resolution:** You can now recover your KVDB from dump files with a single command using the document procedure here - https://docs.portworx.com/concepts/internal-kvdb/#recovery.

**Pull request merged**